client/state/reader/feed-search/reducer.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * External dependencies
+ */
+import { createReducer, keyedReducer } from 'state/utils';
+
+/**
+ * Internal dependencies
+ */
+import { READER_FEED_SEARCH_RECEIVE } from 'state/action-types';
+
+/**
+ * Tracks mappings between queries --> feed results
+ * Here is what the state tree may look like:
+ * feedsearch: {
+		items: {
+			'wordpress tavern': [ feed1, feed2, ],
+			...
+		},
+	}
+ *
+ * @param  {Array} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Array}        Updated state
+ */
+export const items = keyedReducer( 'query', createReducer( [], {
+	[ READER_FEED_SEARCH_RECEIVE ]: ( state, action ) => state.concat( action.payload )
```

Comment:
we should probably key these lists off the feed id? and store the actual feed objects with the feed reducer.